File: backend/src/Core/LSTY.SevenDPanel.Application/Chat/ChatMessage.cs
using System;

namespace LSTY.SevenDPanel.Application.Chat
{
    public enum ChatChannel
    {
        Global,
        Friends,
        Party,
        Whisper,
        Unknown
    }

    public enum ChatSourceKind
    {
        Player,
        Administrator,
        System
    }

    public sealed class ChatMessage
    {
        public required long Sequence { get; init; }
        public required DateTimeOffset OccurredAtUtc { get; init; }
        public required int EntityId { get; init; }
        public string? CrossplatformId { get; init; }
        public required string SenderName { get; init; }
        public required ChatChannel Channel { get; init; }
        public required ChatSourceKind SourceKind { get; init; }
        public required string Message { get; init; }
    }
}

File: frontend/apps/admin/src/features/game-chat/model/chatMessage.ts
export const chatChannels = ['Global', 'Friends', 'Party', 'Whisper', 'Unknown'] as const

export type ChatChannel = typeof chatChannels[number]
export type ChatChannelFilter = 'All' | ChatChannel
export type ChatSourceKind = 'Player' | 'Administrator' | 'System'
export type ChatConnectionStatus = 'connecting' | 'live' | 'reconnecting' | 'stopped'

export interface ChatMessage {
  readonly sequence: number
  readonly occurredAtUtc: string
  readonly entityId: number
  readonly crossplatformId: string | null
  readonly senderName: string
  readonly channel: ChatChannel
  readonly sourceKind: ChatSourceKind
  readonly message: string
}
